lib/create-app.js

```diff
@@ -169,6 +169,23 @@ function initWebId (argv, app, ldp) {
   const useSecureCookies = !!argv.sslKey // use secure cookies when over HTTPS
   const sessionHandler = session(sessionSettings(useSecureCookies, argv.host))
   app.use(sessionHandler)
+  // Reject cookies from third-party applications.
+  // Otherwise, when a user is logged in to their Solid server,
+  // any third-party application could perform authenticated requests
+  // without permission by including the credentials set by the Solid server.
+  app.use((req, res, next) => {
+    const origin = req.headers.origin
+    const userId = req.session.userId
+    // Exception: allow logout requests from all third-party apps
+    if (!argv.host.allowsSessionFor(userId, origin) && !isLogoutRequest(req)) {
```

Comment:
This was a bit hard to review. Is this actually the only change here?

Reply:
This thing is the only real change.